executor/analyze.go

```diff
@@ -206,9 +213,55 @@ func (e *AnalyzeExec) Next(ctx context.Context, req *chunk.Chunk) error {
 			}
 		}
 	}
+	err = e.saveAnalyzeOptsV2()
+	if err != nil {
+		e.ctx.GetSessionVars().StmtCtx.AppendWarning(err)
+	}
 	return statsHandle.Update(e.ctx.GetInfoSchema().(infoschema.InfoSchema))
 }
 
+func (e *AnalyzeExec) saveAnalyzeOptsV2() error {
+	if !variable.PersistAnalyzeOptions.Load() || e.OptionsMap == nil || len(e.OptionsMap) == 0 {
```

Comment:
`!variable.PersistAnalyzeOptions.Load() || len(e.OptionsMap) == 0` is enough.

Reply:
Done